openjdk/mmtkHeap.cpp

```diff
@@ -380,12 +387,16 @@ void MMTkHeap::print_tracing_info() const {
 // Registering and unregistering an nmethod (compiled code) with the heap.
 // Override with specific mechanism for each specialized heap type.
 class MMTkRegisterNMethodOopClosure: public OopClosure {
-  template <class T> void do_oop_work(T* p) {
+  template <class T> void do_oop_work(T* p, bool narrow) {
+    if (UseCompressedOops && !narrow) {
+      guarantee((uintptr_t(p) & (1ull << 63)) == 0, "test");
+      p = (T*) (uintptr_t(p) | (1ull << 63));
```

Comment:
I am being pedantic, but it is usually considered a bad practice to overwrite function arguments.  It is safer to introduce new variables.  It avoids bugs like this one: https://github.com/mmtk/mmtk-core/pull/864#discussion_r1294167769

Reply:
Fixed.